.github/workflows/ci.yml

```diff
@@ -226,22 +226,36 @@ jobs:
 
   ecosystem-test-prefect:
     needs: build-binary-linux
-    name: "ecosystem test | prefect"
+    name: "ecosystem test | ${{ matrix.repo }}"
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - {
+              repo: "prefecthq/prefect",
+              command: "uv pip install --upgrade -e '.[dev]'",
+              python: "3.9",
+            }
+          - {
+              repo: "home-assistant/core",
+              command: "uv pip install -r requirements_all.txt",
+              python: "3.12",
+            }
+      fail-fast: false
     steps:
       - uses: actions/checkout@v4
 
       - uses: actions/setup-python@v5
         with:
-          python-version: "3.9"
+          python-version: ${{ matrix.python }}
 
       - name: "Download binary"
         uses: actions/download-artifact@v4
         with:
           name: uv-linux-${{ github.sha }}
 
       - name: "Clone project"
-        run: gh repo clone prefecthq/prefect
+        run: gh repo clone ${{ matrix.repo }} repo
```

Comment:
We can speed this up with depth 1, `actions/checkout@v4` should do this by default